components/scream/src/diagnostics/tests/field_at_pressure_level_tests.cpp

```diff
@@ -0,0 +1,258 @@
+#include "catch2/catch.hpp"
+
+#include "ekat/ekat_pack_utils.hpp"
+
+#include "diagnostics/field_at_pressure_level.hpp"
+
+#include "share/grid/mesh_free_grids_manager.hpp"
+#include "share/field/field_utils.hpp"
+#include "share/util/scream_setup_random_test.hpp"
+
+namespace scream {
+
+const int packsize = SCREAM_SMALL_PACK_SIZE;
+using Pack         = ekat::Pack<Real,packsize>;
+
+std::shared_ptr<GridsManager>
+create_gm (const ekat::Comm& comm, const int ncols, const int nlevs) {
+
+  const int num_global_cols = ncols*comm.size();
+
+  ekat::ParameterList gm_params;
+  gm_params.set<int>("number_of_global_columns", num_global_cols);
+  gm_params.set<int>("number_of_vertical_levels", nlevs);
+
+  auto gm = create_mesh_free_grids_manager(comm,gm_params);
+  gm->build_grids();
+
+  return gm;
+}
+
+struct PressureBnds
+{
+  Real p_top  = 10000.0;  //  100mb
+  Real p_surf = 100000.0; // 1000mb
+};
+
+std::shared_ptr<GridsManager>
+get_test_gm(const ekat::Comm& io_comm, const int num_gcols, const int num_levs);
+
+std::shared_ptr<FieldManager>
+get_test_fm(std::shared_ptr<const AbstractGrid> grid);
+
+std::shared_ptr<FieldAtPressureLevel>
+get_test_diag(const ekat::Comm& comm, std::shared_ptr<const FieldManager> fm, std::shared_ptr<const GridsManager> gm, const std::string& type, const Real plevel);
+
+Real get_test_pres(const int col, const int lev, const int num_lev, const int num_cols);
+Real get_test_data(const Real pres);
+
+TEST_CASE("field_at_pressure_level_p2")
+{
+  // Get an MPI comm group for test
+  ekat::Comm comm(MPI_COMM_WORLD);
+
+  // Random values to be used in test
+  PressureBnds pressure_bounds;
+  auto engine = scream::setup_random_test(&comm);
+  using RPDF = std::uniform_real_distribution<Real>;
+  RPDF pdf_plev(pressure_bounds.p_top,pressure_bounds.p_surf);
+  
+
+  // Create a grids manager w/ a point grid
+  int ncols = 3;
+  int nlevs = 10;
+  REQUIRE(2*nlevs>=ncols+1);
+  auto gm   = get_test_gm(comm,ncols,nlevs);
+
+  // Create a field manager for testing
+  auto grid = gm->get_grid("Point Grid");
+  auto fm   = get_test_fm(grid);
+  util::TimeStamp t0 ({2022,1,1},{0,0,0});
+
+  // Get a diagnostic factory
+  auto& diag_factory = AtmosphereDiagnosticFactory::instance();
+  diag_factory.register_product("FieldAtPressureLevel",&create_atmosphere_diagnostic<FieldAtPressureLevel>);
+
+  {
+    // Test 1: Take a slice at 500mb (50000Pa) for variable defined at midlevel.
```

Comment:
It seems to me that test 3 and 4 already cover any use case, so 1 and 2 can be removed?